ts/image-occlusion/shapes/ellipse.ts

```diff
@@ -28,20 +28,28 @@ export class Ellipse extends Shape {
     }
 
     toFabric(size: Size): fabric.Ellipse {
-        this.makeAbsolute(size);
-        return new fabric.Ellipse(this);
+        const absolute = this.toAbsolute(size);
+        return new fabric.Ellipse(absolute);
     }
 
-    makeNormal(size: Size): void {
-        super.makeNormal(size);
-        this.rx = xToNormalized(size, this.rx);
-        this.ry = yToNormalized(size, this.ry);
+    toNormal(size: Size): Ellipse {
+        return new Ellipse({
+            ...this,
+            rx: xToNormalized(size, this.rx),
+            ry: yToNormalized(size, this.ry),
+            left: xToNormalized(size, this.left),
```

Comment:
Might be able to shave off the `left` and `top` normalization in this and other `Shape` child classes by more clever use of method inheritance